src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceDao.java

```diff
@@ -183,7 +183,9 @@ public void deleteProject(UUID id) {
             .map(projectRef -> projectRef.projectId)
             .collect(Collectors.toList());
     // mark those that are not in use for delete
-    markProjectsForDelete(projectsToDelete);
+    if (!projectsToDelete.isEmpty()) {
```

Comment:
Can you do this check inside `markProjectsForDelete()` instead? That would avoid making the caller check for something that the callee needs to enforce.